ext/openssl/openssl.c

```diff
@@ -474,6 +474,47 @@ void php_openssl_store_errors()
 }
 /* }}} */
 
+/* {{{ php_openssl_errors_set_mark */
+void php_openssl_errors_set_mark(void) {
+	struct php_openssl_errors *errors;
+	struct php_openssl_errors *errors_mark;
+
+	if (!OPENSSL_G(errors)) {
+		return;
+	}
+
+	if (!OPENSSL_G(errors_mark)) {
+		OPENSSL_G(errors_mark) = pecalloc(1, sizeof(struct php_openssl_errors), 1);
```

Comment:
This needs to be freed in `PHP_GSHUTDOWN_FUNCTION`.

Reply:
I've also added 
```
openssl_globals->errors_mark = NULL;
```
to `PHP_GINIT_FUNCTION` (to mirror what is done with `openssl_globals->errors`)